### ISSUE TITLE
Server doesn't shut down after server crash

### ISSUE THREAD

**Issue by Zekrom64** [SpongePowered/SpongeForge#2614]
**I am currently running**
- SpongeForge version: spongeforge-1.12.2-2768-7.1.5-RC3520
- Forge version: 14.23.5.2781
- Java version: 1.8.0_71
- Operating System: openSUSE Leap 42.3

- Plugins/Mods:
**Plugins** Minecraft, Minecraft Coder Pack, SpongeAPI, Sponge, CatClearLag, ClearMob, CmdControl, CmdScheduler, DeathPenalty Reloaded, EconomyLite, ForgedPrms, GriefPrevention, LuckPerms, MBDiscordLink-API7, MMCReboot, MMCRules, MagiBridge, Nucleus, PayDay, PayRespects, Plan, Web-API
**Modpack** [Enigmatica 2: Expert](https://minecraft.curseforge.com/projects/enigmatica2expert)

**Issue Description**
Running a large modpack on a server with several active players the server will occasionally crash. This is ok as we have a script that will automatically restart the server once the process exits. More recently when the server has crashed the process fails to exit and we have to manually kill the process for the script to kick in. In the server console it will say that the server is shutting down but it will not close, the console becomes unresponsive and continues printing messages.

Most messages seem to be from Sponge's task scheduler about exceptions thrown running a task, leading me to believe that SpongeForge or a plugin isn't stopping all of its threads when the server crashes and keeping the process alive. The server is on a remote machine making it difficult to use tools like VisualVM so I don't have a complete dump of all the threads.

[crash-2019-02-16_13.53.36-server.txt](https://github.com/SpongePowered/SpongeForge/files/2872301/crash-2019-02-16_13.53.36-server.txt)
[debug-1.log](https://gist.github.com/Zekrom64/61a3c506e78e2c04a78d2271679de682) (Truncated to end, file is 49MB uncompressed)

**Comment by Zidane** [comment]
This is setting something off the main thread:

	at lumien.randomthings.tileentity.TileEntityAdvancedRedstoneTorch.increaseSignalStrengthOff(TileEntityAdvancedRedstoneTorch.java:80) ~[TileEntityAdvancedRedstoneTorch.class:?]

Can't do that.

**Comment by phit** [comment]
in `config/sponge/global.conf`, set `modules.broken-mod=true`. Then, under the `broken-mods` section, add `randomthings` to `broken-network-handler-mods`, so `broken-network-handler-mods=[randomthings]`

that should stop the crash itself, though that doesn't explain the server not shutting down fully

**Comment by Zekrom64** [comment]
We managed to get a JMX connection to the server when it hung, the 'Server thread' seems to be stuck waiting on `System.exit`. Additionally, 'Server Shutdown Thread' is still running and calling code for the WebAPI plugin.

**Comment by dandhy12** [comment]
Pada tanggal Rab, 20 Feb 2019 22:37 Zekrom_64 <notifications@github.com
menulis:

**Comment by phit** [comment]
for the actual issue here i don't see anything pointing at Sponge, if you find a way to reproduce feel free to make a new issue

as for the random things issue, i opened a ticket on the mod tracker https://github.com/lumien231/Random-Things/issues/416
